aws-iot-softwarepackageversion/src/main/java/software/amazon/iot/softwarepackageversion/CreateHandler.java

```diff
@@ -0,0 +1,92 @@
+    private static final String CALL_GRAPH = "AWS-IoT-SoftwarePackageVersion::Create";
+    private static final int MAX_PACKAGE_NAME_LENGTH = 128;
+    private Logger logger;
+    private String clientToken;
+
+    @Override
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<IotClient> proxyClient,
+            final Logger logger) {
+
+        this.logger = logger;
+        this.clientToken = request.getClientRequestToken();
+
+        final ResourceModel resourceModel = request.getDesiredResourceState();
+
+        return ProgressEvent.progress(resourceModel, callbackContext)
+                .then(progress ->
+                        proxy.initiate(CALL_GRAPH, proxyClient, resourceModel, callbackContext)
+                                .translateToServiceRequest(Translator::translateToCreateRequest)
+                                .makeServiceCall(this::createResource)
+                                .progress())
+                .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
+    }
+
+    /**
+     * Implement client invocation of the create request through the proxyClient, which is already initialised with
+     * caller credentials, correct region and retry settings
+     *
+     * @param createPackageVersionRequest     the aws service request to create a resource
+     * @param proxyClient the aws service client to make the call
+     * @return awsResponse create resource response
+     */
+    private CreatePackageVersionResponse createResource(
+            final CreatePackageVersionRequest createPackageVersionRequest,
+            final ProxyClient<IotClient> proxyClient) {
+        // A little bit messy, but need to append the idempotency token which is not a property of the ResourceModel
+        CreatePackageVersionRequest requestWithClientToken = CreatePackageVersionRequest.builder()
+                .packageName(createPackageVersionRequest.packageName())
+                .versionName(createPackageVersionRequest.versionName())
+                .description(createPackageVersionRequest.description())
+                .attributes(createPackageVersionRequest.attributes())
+                .tags(createPackageVersionRequest.tags())
+                //.clientToken(this.clientToken)
+                .build();
+        try {
+            final CreatePackageVersionResponse createPackageVersionResponse = proxyClient.injectCredentialsAndInvokeV2(
+                    requestWithClientToken, proxyClient.client()::createPackageVersion);
+            logger.log(String.format("%s [%s] successfully created.",
+                    ResourceModel.TYPE_NAME, createPackageVersionRequest.packageName()));
```

Comment:
Can we log the accountId

Reply:
yes, I'll create an action item for that.

Reply:
added logging